tests/PHPStan/Rules/Methods/FriendMethodCallRuleTest.php

```diff
@@ -0,0 +1,65 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Methods;
+
+use PHPStan\Rules\Rule;
+use PHPStan\Type\FileTypeMapper;
+
+/**
+ * @extends \PHPStan\Testing\RuleTestCase<FriendMethodCallRule>
+ */
+class FriendMethodCallRuleTest extends \PHPStan\Testing\RuleTestCase
+{
+
+	protected function getRule(): Rule
+	{
+		return new FriendMethodCallRule(self::getContainer()->getByType(FileTypeMapper::class));
+	}
+
+	public function testFriendMethodCall(): void
+	{
+		$this->analyse([ __DIR__ . '/data/friend-method-call.php'], [
+			[
+				'You may not not call FriendMethodCallTest\Foo::friendWithEntireBar() from the global scope as the method lists allowed callers through friends.',
+				6,
+			],
+			[
+				'FriendMethodCallTest\Bar::notBaz() may not call FriendMethodCallTest\Foo::friendWithBarBaz() as it is not listed as a friend.',
+				48,
+			],
+			[
+				'FriendMethodCallTest\Bar::notBaz() may not call FriendMethodCallTest\FooFoo::friendWithBarBaz() as it is not listed as a friend.',
+				53,
```

Comment:
I'm having a hard time figuring out why this violation doesn't pop up. I'll try to debug some more, pushing this already for other changes

Reply:
I'm suspecting `$docComment = $method->getDocComment();` that this returns `null` and then rule is exiting early due to this. Is there a way to get a `ResolvedPhpDocBlock` for a method that would include all parents resolved?

Reply:
@ondrejmirtes any hint please?

Reply:
Sorry, didn't notice you have a question here. Try something like this:

```diff
--- a/src/Reflection/Php/PhpClassReflectionExtension.php
+++ b/src/Reflection/Php/PhpClassReflectionExtension.php
@@ -498,6 +498,7 @@ class PhpClassReflectionExtension
 					$methodReflection->getName(),
 					$positionalParameterNames
 				);
+				$docComment = $resolvedPhpDoc->getPhpDocString();
 				$phpDocBlockClassReflection = $declaringClass;
 			}
 		} else {
```

Although I'm not sure + I don't know why the `getPhpDocString()` isn't nullable, it might be returning an empty string instead (which we should change to `null` here).

Reply:
This definitely won't work, but at least I showed you the place that needs fixing :)

Reply:
(I'm in favor of having the inherited phpDoc be returned from a child's method getDocComment.)